includes/class-sensei-learner.php

```diff
@@ -111,6 +114,63 @@ public function delete_all_user_activity( $user_id = 0 ) {
 		return $dataset_changes;
 	}
 
+	/**
+	 * Remove duplicate progress comments to mitigate duplicate enrollment issue.
+	 *
+	 * Hooked into sensei_log_activity_after
+	 *
+	 * @since 3.7.0
+	 * @access private
+	 *
+	 * @param array $args
+	 */
+	public function remove_duplicate_progress( $args ) {
+		add_action(
+			'shutdown',
+			function() use ( $args ) {
+				global $wpdb;
+
+				// Get progress comments, but the first one, which match the context conditions (they should not exist).
+				// phpcs:disable WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching, WordPress.DB.PreparedSQL.NotPrepared
+				$comment_ids = $wpdb->get_col(
+					$wpdb->prepare(
+						"SELECT comment_ID
+							FROM $wpdb->comments
+							WHERE comment_post_ID = %d
+								AND user_id = %d
+								AND comment_type = %s
+							ORDER BY comment_ID
```

Comment:
Would it make more sense to order by `comment_date`?

Reply:
Considering we want to keep the first comment progress only, I think the `comment_ID` would be safer because the `comment_date` changes when we change the progress. So depending on the issue origin, we could remove a comment with some progress, instead of the new one.

Reply:
I was wondering if there was the ability to make comment IDs random instead of necessarily auto-incrementing. Doing a little bit of Googling, it seems that this probably does not happen (or at least it is very unlikely). So I think this is ok 👍